beluga_benchmark/docs/reports/2023-09-02/REPORT.md

```diff
@@ -0,0 +1,114 @@
+- `beluga_amcl` (beam, non-multithreaded) uses [beam_params.yaml](beam_params.yaml)
+- `beluga_amcl` (beam, multithreaded) uses [beam_params_par.yaml](beam_params_par.yaml)
+
+Except for the multithreading and sensor model parameters, the configuration on all of the files is identical.
+
+## Recorded metrics
+
+The following metrics were recorded during each run:
+
+- RSS (Resident Set Size), amount of memory alloated to the process. Measured in megabytes.
+- CPU usage. Measured in percentage of the total CPU usage.
+- rms APE (root-mean-squared Absolute Pose Error) statistics. In meters.
+
+## Results
+
+#### Beluga vs. Nav2 AMCL using Likelihood Field sensor model
+
+In the following plot the results of the benchmark are shown for all three of the tested configurations. The vertical scale is logarithmic to better show the differences between the configurations throughout the whole range of particle counts.
+
+![Beluga Seq vs Beluga Par vs. Nav2 AMCL with Likelihood Field Sensor Model](likelihood_beluga_vs_beluga_vs_amcl.png)
+
+Comments on the results:
+
+- The memory usage of `beluga_amcl` (both configurations) is significantly lower than that of `nav2_amcl`.
+- Both `beluga_amcl` configurations perform at a lower CPU usage level than `nav2_amcl` when using the `likelihood` sensor model.
+- Above $50k$ particles `nav2_amcl` begins to saturate the CPU and its APE metrics begin to deteriorate significantly, while `beluga_amcl`'s remain stable. For particle conts below $50k$ the APE metrics of both `beluga_amcl` and `nav2_amcl` are similar.
+
+#### Beluga vs. Nav2 AMCL using Beam sensor model
+
+In the following plot the results of the benchmark are shown for all three of the tested configurations when using the Beam Sensor model. The vertical scale is logarithmic to better show the differences between the configurations throughout the whole range of particle counts.
+
+![Beluga Seq vs Beluga Par vs. Nav2 AMCL with Beam Sensor Model](beam_beluga_vs_beluga_vs_amcl.png)
+
+Comments on the results:
+
+- `beluga_amcl` in both multithreaded and non-multithreaded configurations uses significantly less memory than `nav2_amcl`.
+- Both `beluga_amcl` configurations perform at a lower CPU usage level than `nav2_amcl` when using the `beam` sensor model.
+- The APE performance of both multithreaded and non-multithreaded `beluga_amcl` is similar to that of `nav2_amcl` throughout the whole range of particle counts.
+
+## Conclusions
+
+- `beluga_amcl`'s memory usage is significantly lower than that of `nav2_amcl` in all configurations.
+- `beluga_amcl`'s CPU usage is consistently lower than that of `nav2_amcl` in all configurations, scaling better with particle count.
+- `beluga_amcl`'s APE performance is similar to that of `nav2_amcl` for lower particle counts. For particles counts higher than $50k$, `nav2_amcl` single-threaded process saturates the CPU and its APE performance drops significantly, while `beluga_amcl` remains stable.
```

Comment:
I see some amount of saturation for the single-threaded beam model in `beluga_amcl`.

Reply:
There is a limit though to how much work you can push through a single core. I see the somewhat exponential relationship between CPU usage and particle count as the bigger threat.

Reply:
Right! I think that's very reasonable. I just mentioned it because the text says `beluga_amcl` remains stable for particle counts higher than 50k with the beam sensor model, and that doesn't seem to be the case according to the plots.